src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Protocol/JsonHelpers.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.CodeAnalysis.Razor.Protocol;
+
+internal static class JsonHelpers
+{
+    private const string s_convertedFlag = "__convertedFromJsonElement";
+
+    /// <summary>
+    /// Normalizes data from JsonElement to JObject as thats what we expect to process
+    /// </summary>
+    internal static object? TryConvertFromJsonElement(object? data)
```

Comment:
Why do these helpers take `object?` and return `object?`? I'm not seeing the reason why they can't just refer to `JsonElement` and `JObject`.

Reply:
Oh, I see. You're being a bit tricky in the custom message targets below. That makes sense.

Reply:
Yeah, I wanted this to be easy to call mainly. In another PR I've changing it to convert to JsonElement, because it has to convert to what we use, but also wanted Roslyn and Web Tools to be free to move independently and without us having to do any dual insertion stuff.